doc/src/manual/running-external-programs.md

```diff
@@ -60,6 +60,56 @@ julia> open(`less`, "w", STDOUT) do io
 3
 ```
 
+## Cmd Object
+
+Normally, to create a `Cmd` object in the first place, one uses backticks, e.g.
+
+```jldoctest
+julia> Cmd(`echo "Hello world"`, ignorestatus=true, detach=false)
+`echo 'Hello world'`
+```
+
+## Execution Environment
+
+You can control a `Cmd`'s execution environment by one of several means. First, constructing a `Cmd` by explicitly calling the `Cmd` type constructor (rather than implicitly using solely backticks as illustrated above) allows you to specify several aspects of the `Cmd`'s execution environment via keyword arguments. For example, the `dir` keyword provides control over the `Cmd`'s working directory,
+
+```jldoctest
+julia> run(Cmd(`pwd`, dir = "/"))
+/
+```
+
+and the `env` keyword allows you to set execution environment variables,
+
+```jldoctest
+julia> run(Cmd(`bash -c "echo foo \$HOWLONG"`, env = ("HOWLONG" => "ever!",)))
+foo ever!
+```
+
+See [`Cmd`](@ref) for additional keyword arguments. [`setenv`](@ref) provides another means for setting Cmd execution environment variables,
+
+```jldoctest
+julia> run(setenv(`bash -c "echo foo \$HOWLONG"`, ("HOWLONG" => "ever!",)))
+foo ever!
+```
+
+and similarly [`withenv`](@ref) for modifying rather than replacing the execution environment,
+
+```jldoctest
+julia> withenv(() -> run(`bash -c "echo foo \$HOWLONG"`), "HOWLONG" => "ever!")
+foo ever!
+```
+
+which commonly appears in do-block form,
+
+```jldoctest
+julia> withenv("HOWLONG" => "ever!") do
+           run(`bash -c "echo foo \$HOWLONG"`)
+       end
+foo ever!
+```
+
+For more information on pipelining of commands see [`pipeline`](@ref).
```

Comment:
Why mention pipelining at this stage? Seems disjoint from the immediately adjacent material, and the reader will hit pipelining naturally two sections down.